### ISSUE TITLE
CPU usage issue with aardvark-dns on Fedora 35 running podman 4.0.0-rc4

### ISSUE THREAD

**Issue by giannisalinetti** [containers/podman#13171]
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind bug

**Description**

**Steps to reproduce the issue:**
1. Create a Vagrant machine on with KVM hypervisor using the following Vagrantfile:
```
Vagrant.configure("2") do |config|
  config.vm.box = "fedora/35-cloud-base"
  config.vm.hostname = "f35.example.com"
  config.vm.provider "libvirt" do |lv|
    lv.memory = 8192
    lv.cpus = 2
  end
end
```

2. Install Podman 4.0.0-rc4 from rhcontainerbot/podman4 copr repositories.
```
sudo dnf copr enable rhcontainerbot/podman4
sudo dnf install podman
```
If Podman is already installed upgrade to the latest release candidate:
```
sudo dnf upgrade podman
```

If CNI network was previously used it is necessary to run a `podman system reset` to clean up all previsou configurations and start using netavark and aadvark-dns.

3. Create a new network:
```
sudo podman network create test-net1
```

4. Create a test container:
```
sudo podman run -d --net test-net1 --name test-c1 docker.io/library/httpd:latest
```

After the container is started the aardvark-dns process is started to serve the network resolution for the `test-net` network. Aardvark stores its configuration for rootful containers under `/run/containers/networks/aardvark-dns`.

**Describe the results you received:**
The container networking seems to work correctly but the aardvark-dns process eats up to 100% of CPU on a dual thread Vagrant virtual machine running with KVM hypervisor.

```
$ top -b -p $(pgrep aardvark-dns) 1
top - 23:33:17 up  4:26,  1 user,  load average: 2.68, 2.76, 2.11
Tasks:   1 total,   0 running,   1 sleeping,   0 stopped,   0 zombie
%Cpu0  :100.0 us,  0.0 sy,  0.0 ni,  0.0 id,  0.0 wa,  0.0 hi,  0.0 si,  0.0 st     %Cpu1  :  0.0 us,  0.0 sy,  0.0 ni,100.0 id,  0.0 wa,  0.0 hi,  0.0 si,  0.0 st
MiB Mem :   7940.0 total,   6180.4 free,    226.0 used,   1533.6 buff/cache
MiB Swap:   7940.0 total,   7940.0 free,      0.0 used.   7446.3 avail Mem 

    PID USER      PR  NI    VIRT    RES    SHR S  %CPU  %MEM     TIME+ COMMAND
  14596 root      20   0  682728   2660   1972 S 100.0   0.0  65:26.18 aardvark-dns
```

Adding more networks and containers the resulting CPU time increases up to the CPU saturation limit.

**Describe the results you expected:**
The aardvark-dns process runs within acceptable CPU comsuption rates.

**Additional information you deem important (e.g. issue happens only occasionally):**
Issue happens constantly.

The issue was already opened and closed on aardvark-dns project.
https://github.com/containers/aardvark-dns/issues/46

The following PR has fixed the issue but this is not yet implemented in the current aardvark-dns package installed with the Podman release in Fedora:
https://github.com/containers/aardvark-dns/pull/48

**Output of `podman version`:**
```
Client:       Podman Engine
Version:      4.0.0-rc4
API Version:  4.0.0-rc4
Go Version:   go1.16.13

Built:      Fri Feb  4 15:25:35 2022
OS/Arch:    linux/amd64
```

**Output of `/usr/libexec/podman/aardvark-dns --version`:**
```
/usr/libexec/podman/aardvark-dns --version
aardvark-dns 1.0.0-rc1
```

**Output of `podman info --debug`:**

```
host:
  arch: amd64
  buildahVersion: 1.24.0
  cgroupControllers:
  - memory
  - pids
  cgroupManager: systemd
  cgroupVersion: v2
  conmon:
    package: conmon-2.1.0-2.fc35.x86_64
    path: /usr/bin/conmon
    version: 'conmon version 2.1.0, commit: '
  cpus: 2
  distribution:
    distribution: fedora
    variant: cloud
    version: "35"
  eventLogger: journald
  hostname: f35.example.com
  idMappings:
    gidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
    uidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
  kernel: 5.14.10-300.fc35.x86_64
  linkmode: dynamic
  logDriver: journald
  memFree: 6475481088
  memTotal: 8325742592
  networkBackend: netavark
  ociRuntime:
    name: crun
    package: crun-1.4.2-1.fc35.x86_64
    path: /usr/bin/crun
    version: |-
      crun version 1.4.2
      commit: f6fbc8f840df1a414f31a60953ae514fa497c748
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +CRIU +YAJL
  os: linux
  remoteSocket:
    path: /run/user/1000/podman/podman.sock
  security:
    apparmorEnabled: false
    capabilities: CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: true
    seccompEnabled: true
    seccompProfilePath: /usr/share/containers/seccomp.json
    selinuxEnabled: true
  serviceIsRemote: false
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: slirp4netns-1.1.12-2.fc35.x86_64
    version: |-
      slirp4netns version 1.1.12
      commit: 7a104a101aa3278a2152351a082a6df71f57c9a3
      libslirp: 4.6.1
      SLIRP_CONFIG_VERSION_MAX: 3
      libseccomp: 2.5.0
  swapFree: 8325689344
  swapTotal: 8325689344
  uptime: 4h 27m 45.9s (Approximately 0.17 days)
plugins:
  log:
  - k8s-file
  - none
  - passthrough
  - journald
  network:
  - bridge
  - macvlan
  volume:
  - local
registries:
  search:
  - registry.fedoraproject.org
  - registry.access.redhat.com
  - docker.io
  - quay.io
store:
  configFile: /home/vagrant/.config/containers/storage.conf
  containerStore:
    number: 2
    paused: 0
    running: 0
    stopped: 2
  graphDriverName: overlay
  graphOptions: {}
  graphRoot: /home/vagrant/.local/share/containers/storage
  graphStatus:
    Backing Filesystem: btrfs
    Native Overlay Diff: "true"
    Supports d_type: "true"
    Using metacopy: "false"
  imageCopyTmpDir: /var/tmp
  imageStore:
    number: 2
  runRoot: /run/user/1000/containers
  volumePath: /home/vagrant/.local/share/containers/storage/volumes
version:
  APIVersion: 4.0.0-rc4
  Built: 1643988335
  BuiltTime: Fri Feb  4 15:25:35 2022
  GitCommit: ""
  GoVersion: go1.16.13
  OsArch: linux/amd64
  Version: 4.0.0-rc4
```

**Package info (e.g. output of `rpm -q podman` or `apt list podman`):**

```
rpm -q podman
podman-4.0.0-0.5.rc4.fc35.x86_64
```

**Have you tested with the latest version of Podman and have you checked the Podman Troubleshooting Guide? (https://github.com/containers/podman/blob/main/troubleshooting.md)**

Yes

**Additional environment details (AWS, VirtualBox, physical, etc.):**
Vagrant with LIbvirt KVM Provider
VM OS: Fedora 35
Host OS: Fedora 34

**Comment by giannisalinetti** [comment]
Tested a new build of aardvark-dns with the merged PR https://github.com/containers/aardvark-dns/pull/48 and the issue appears to be solved.

**Steps to reproduce:**
1. Clone the aardvark-dns repository
```
git clone https://github.com/containers/aardvark-dns.git
```

2. Install necessary dependecies to build
```
sudo dnf install -y make rust cargo
```

3. Build the new binary
```
cd aardvark-dns
make
```

4. Copy the new binary under /usr/libexec/podman
```
cp ./bin/aardvark-dns /usr/libexec/podman
```

5. Test by running a new container
```
sudo podman run -d --net test-net1 --name test-c1 docker.io/library/httpd:latest
```

**Expected results:**
CPU compusumption should be minimal for the aardvark-dns process.
```
top -b -p $(pgrep aardvark-dns)
top - 00:14:24 up  5:07,  1 user,  load average: 0.00, 0.35, 0.91
Tasks:   1 total,   0 running,   1 sleeping,   0 stopped,   0 zombie
%Cpu(s):  0.0 us,  0.0 sy,  0.0 ni,100.0 id,  0.0 wa,  0.0 hi,  0.0 si,  0.0 st
MiB Mem :   7940.0 total,   5214.3 free,    237.9 used,   2487.9 buff/cache
MiB Swap:   7940.0 total,   7940.0 free,      0.0 used.   7419.6 avail Mem 

    PID USER      PR  NI    VIRT    RES    SHR S  %CPU  %MEM     TIME+ COMMAND
  22162 root      20   0  479944   2436   2016 S   0.0   0.0   0:00.00 aardvark-dns
  ```

**Comment by flouthoc** [comment]
> Tested a new build of aardvark-dns with the merged PR [containers/aardvark-dns#48](https://github.com/containers/aardvark-dns/pull/48) and the issue appears to be solved.
> 
> **Steps to reproduce:**
> 
>     1. Clone the aardvark-dns repository
> 
> 
> ```
> git clone https://github.com/containers/aardvark-dns.git
> ```
> 
>     2. Install necessary dependecies to build
> 
> 
> ```
> sudo dnf install -y make rust cargo
> ```
> 
>     3. Build the new binary
> 
> 
> ```
> cd aardvark-dns
> make
> ```
> 
>     4. Copy the new binary under /usr/libexec/podman
> 
> 
> ```
> cp ./bin/aardvark-dns /usr/libexec/podman
> ```
> 
>     5. Test by running a new container
> 
> 
> ```
> sudo podman run -d --net test-net1 --name test-c1 docker.io/library/httpd:latest
> ```
> 
> **Expected results:** CPU compusumption should be minimal for the aardvark-dns process.
> 
> ```
> top -b -p $(pgrep aardvark-dns)
> top - 00:14:24 up  5:07,  1 user,  load average: 0.00, 0.35, 0.91
> Tasks:   1 total,   0 running,   1 sleeping,   0 stopped,   0 zombie
> %Cpu(s):  0.0 us,  0.0 sy,  0.0 ni,100.0 id,  0.0 wa,  0.0 hi,  0.0 si,  0.0 st
> MiB Mem :   7940.0 total,   5214.3 free,    237.9 used,   2487.9 buff/cache
> MiB Swap:   7940.0 total,   7940.0 free,      0.0 used.   7419.6 avail Mem 
> 
>     PID USER      PR  NI    VIRT    RES    SHR S  %CPU  %MEM     TIME+ COMMAND
>   22162 root      20   0  479944   2436   2016 S   0.0   0.0   0:00.00 aardvark-dns
> ```

Yes this should not be in upstream anymore. But its weird that you had to manually pull and build from upstream.

@baude @lsm5 is copr not reflecting upstream commit for aardvark.

**Comment by flouthoc** [comment]
I'm closing this issue since copr reflecting latest upstream is a different issue.
